### ISSUE TITLE
Eigenvalues not sorted for almost perfectly symmetric matrix

### ISSUE THREAD

**Issue by jeff-regier** [JuliaLang/julia#17093]
I'm not sure whether this qualifies as a bug, though the behavior seems undesirable. Below `A` is a matrix that is almost exactly symmetric. `eigvals(A)` returns the singular values without sorting them. For perfectly symmetric matrices (e.g., `B` below), the eigenvalues are always sorted. It seems like `eigvals(A)` should sort them too, since due to numerical issues, even symmetric matrices are often not exactly symmetric.

```
julia> A = [2.319564160285209 20.613046913136856 -0.5068584177942504
            20.613046913136856 201.80977596527438 -1.2031203195074227
            -0.5068584177942503 -1.203120319507423 11.188307566885028];

julia> eigvals(A)  # these singular values are not sorted
3-element Array{Float64,1}:
 203.926   
   0.198637
  11.1935  

julia> norm(A - A')  # though A is almost perfectly symmetric
2.482534153247273e-16

julia> B = (A + A') / 2.  # but if A is symmetrized
3x3 Array{Float64,2}:
  2.31956    20.613    -0.506858
 20.613     201.81     -1.20312 
 -0.506858   -1.20312  11.1883  

julia> eigvals(B)  # then they are sorted
3-element Array{Float64,1}:
   0.198637
  11.1935  
 203.926   
```

**Comment by stevengj** [comment]
If you know you should have a symmetric matrix, you are well-advised to make it perfectly symmetric so that `eigvals` can use a more efficient algorithm.   Maybe the lack of sorting here is a useful clue.

**Comment by andreasnoack** [comment]
...and remember that you can simply do `eigvals(Symmetric(A))` to enforce that a symmetric solver is being used.

**Comment by jeff-regier** [comment]
I'll use `Symmetric` in my own program, thanks for suggestion.

Returning eigenvalues out-of-order seems to me like a pretty cryptic way to alert users that a matrix isn't perfectly symmetric. Seems like tiny perturbations of real-valued input shouldn't drastically change the output, if possible.

**Comment by KristofferC** [comment]
There has been discussion about putting tolerances the symmetry checks but nothing concrete has Viber out of it.

**Comment by jiahao** [comment]
Ref: discussion in #10298 about approximate symmetry checks in `ishermitian` and `issymmetric`

**Comment by stevengj** [comment]
Closed by #21598.
